db/migrate/20230227213252_add_recorded_at_to_recordings.rb

```diff
@@ -0,0 +1,24 @@
+# BigBlueButton open source conferencing system - http://www.bigbluebutton.org/.
+#
+# Copyright (c) 2022 BigBlueButton Inc. and by respective authors (see below).
+#
+# This program is free software; you can redistribute it and/or modify it under the
+# terms of the GNU Lesser General Public License as published by the Free Software
+# Foundation; either version 3.0 of the License, or (at your option) any later
+# version.
+#
+# Greenlight is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with Greenlight; if not, see <http://www.gnu.org/licenses/>.
+
+# frozen_string_literal: true
+
+class AddRecordedAtToRecordings < ActiveRecord::Migration[7.0]
+  def change
+    # rubocop:disable Rails/NotNullColumn
```

Comment:
Is this a new cop? It complains that we are adding `null: false` without providing default value but it does not seem to be a problem elsewhere.

Reply:
Hmm this is a tricky scenario. essentially, we're adding a new column that can't be null without specifying a default value, which is problematic. Maybe drop the null: false?

Reply:
It should not be a problem if users do `migrate:with_data`, which will populate the column with the `created_at` value. No?
I think `null: false` should stay.

Reply:
data_migrations are schema_migrations are independent of each other. When the schema_migration runs, it adds a new column `recorded_at` but doesn't know what the value of the column should be. Since the column can't be null, it's complaining

Reply:
If you wan't to keep the `null: false` we need to define a default value

Reply:
We could set the default value to `created_at` instead of doing it in the data_migration? And keep the `null: false`

Reply:
I don't believe you can reference another column in that way

Reply:
> data_migrations are schema_migrations are independent of each other. When the schema_migration runs, it adds a new column `recorded_at` but doesn't know what the value of the column should be. Since the column can't be null, it's complaining

I don't get what the difference is here compared to any other table we created with a column that has null false? Wouldn't they all be this same situation?  Why is only this one complaining

Reply:
it's only for `add_column` not for `create_table`. There's only 3 other cases of `add_column` and they all have default values defined